src/Arcus.WebApi.Jobs/AzureServiceBusCloudEventSubscriptionMessagePump.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Arcus.Messaging.Abstractions;
+using Arcus.Messaging.Pumps.ServiceBus;
+using CloudNative.CloudEvents;
+using GuardNet;
+using Microsoft.Azure.ServiceBus;
+using Microsoft.Azure.ServiceBus.Management;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+
+namespace Arcus.WebApi.Jobs
+{
+    /// <summary>
+    /// Representing a Azure ServiceBus Topic message pump that will create and delete a ServiceBus Topic subscription during the lifetime of the pump.
+    /// </summary>
+    public abstract class AzureServiceBusCloudEventSubscriptionMessagePump : AzureServiceBusMessagePump<CloudEvent>
```

Comment:
I presume this will be our generic background job from #96, right?

In that case, I'd simply call it `CloudEventBackgroundJob` or so? From a function perspective it's just a background job that processes CloudEvents, the fact that it uses Service Bus message pump is more of an implementation detail, no?

Reply:
Ah, yes. Cool. Because I didn't quite find a dev-friendly name for it.

Reply:
I'm just thinking out loud, does it make sense? 😅

Reply:
Thanks for renaming

Reply:
Jep, renamed.